diff --git a/PicatchUsFB/www/js/app.js b/PicatchUsFB/www/js/app.js
index b5a8d30..7c6925a 100644
--- a/PicatchUsFB/www/js/app.js
+++ b/PicatchUsFB/www/js/app.js
@@ -45,15 +45,6 @@ angular.module('starter', ['ionic', 'ngLocalStorage','ui.router', 'ngOpenFB', 'n
       controller: 'FirstUseController'
     })
 
-    .state('editPicture', {
-      url: "/editPicture",
-      templateUrl:'pages/editPicture/editPicture.html',
-      controller:'EditPictureController',
-      params: {
-        imageURI: null
-      }
-    })
-
     .state('home', {
       url: '/home',
       abstract: true,
@@ -71,6 +62,19 @@ angular.module('starter', ['ionic', 'ngLocalStorage','ui.router', 'ngOpenFB', 'n
       }
     })
 
+    .state('home.editPicture', {
+      url: "/editPicture",
+      views: {
+        'editPicture-tab': {
+          templateUrl:'pages/editPicture/editPicture.html',
+          controller:'EditPictureController'
+        }
+      },
+      params: {
+        imageURI: null
+      }
+    })
+
     .state('home.userEvents', {
       url: "/userEvents",
       views: {
@@ -93,3 +97,4 @@ angular.module('starter', ['ionic', 'ngLocalStorage','ui.router', 'ngOpenFB', 'n
 })
 
 var app = angular.module('starter.controllers', ['starter.filters']);
+var service = angular.module('starter.services', []);
diff --git a/PicatchUsFB/www/pages/home/HomeController.js b/PicatchUsFB/www/pages/home/HomeController.js
index 6114eb5..8f28108 100644
--- a/PicatchUsFB/www/pages/home/HomeController.js
+++ b/PicatchUsFB/www/pages/home/HomeController.js
@@ -6,7 +6,7 @@ app.controller('HomeController',function ($scope, $state){
         });
 
         function onSuccess(imageURI) {
-			$state.go('editPicture', {imageURI: imageURI});
+			$state.go('home.editPicture', {imageURI: imageURI});
         }
 
         function onFail(message) {
diff --git a/PicatchUsFB/www/pages/home/main.html b/PicatchUsFB/www/pages/home/main.html
index 4093f40..2b27f62 100644
--- a/PicatchUsFB/www/pages/home/main.html
+++ b/PicatchUsFB/www/pages/home/main.html
@@ -5,7 +5,7 @@
         </ion-tab>
 
         <ion-tab icon="ion-camera" ng-click="takePicture()">
-          <ion-nav-view name="editPicture-tab"></ion-nav-view>
+          <ion-nav-view name="editPicture-tab" class="tabs-item-hide"></ion-nav-view>
         </ion-tab>
 
         <ion-tab icon="ion-images" ui-sref="home.userEvents">
